module/executiondatasync/optimistic_sync/pipeline/core.go

```diff
@@ -0,0 +1,305 @@
+)
+
+// ErrAbandoned is returned when the core is used after it has been abandoned.
+var ErrAbandoned = errors.New("core is abandoned")
+
+// TODO: DefaultTxResultErrMsgsRequestTimeout should be configured in future PR`s
+
+const DefaultTxResultErrMsgsRequestTimeout = 5 * time.Second
+
+// workingData encapsulates all components and temporary storage
+// involved in processing a single block's execution data. When processing
+// is complete or abandoned, the entire workingData can be discarded.
+type workingData struct {
+	// Temporary in-memory caches
+	inmemRegisters       *unsynchronized.Registers
+	inmemEvents          *unsynchronized.Events
+	inmemCollections     *unsynchronized.Collections
+	inmemTransactions    *unsynchronized.Transactions
+	inmemResults         *unsynchronized.LightTransactionResults
+	inmemTxResultErrMsgs *unsynchronized.TransactionResultErrorMessages
+
+	// Active processing components
+	execDataRequester             requester.ExecutionDataRequester
+	txResultErrMsgsRequester      tx_error_messages.Requester
+	txResultErrMsgsRequestTimeout time.Duration
+	indexer                       *indexer.InMemoryIndexer
+	blockPersister                *persisters.BlockPersister
+	registersPersister            *persisters.RegistersPersister
+
+	// Working data
+	executionData       *execution_data.BlockExecutionDataEntity
+	txResultErrMsgsData []flow.TransactionResultErrorMessage
+}
+
+var _ optimistic_sync.Core = (*Core)(nil)
+
+// Core implements the core logic for execution data processing. It exposes methods for each of the
+// processing steps, which must be called sequentially in the order: Download, Index, Persist.
+// Abandon may be called at any time to abort processing and cleanup working data.
+// The Core instance cannot be used after Abandon is called, and will return ErrAbandoned.
+type Core struct {
+	log zerolog.Logger
+	mu  sync.Mutex
+
+	workingData *workingData
+
+	executionResult *flow.ExecutionResult
+	header          *flow.Header
+}
+
+// NewCore creates a new Core with all necessary dependencies
+func NewCore(
+	logger zerolog.Logger,
+	executionResult *flow.ExecutionResult,
+	header *flow.Header,
+	execDataRequester requester.ExecutionDataRequester,
+	txResultErrMsgsRequester tx_error_messages.Requester,
+	txResultErrMsgsRequestTimeout time.Duration,
+	persistentRegisters storage.RegisterIndex,
+	persistentEvents storage.Events,
+	persistentCollections storage.Collections,
+	persistentResults storage.LightTransactionResults,
+	persistentTxResultErrMsg storage.TransactionResultErrorMessages,
+	latestPersistedSealedResult storage.LatestPersistedSealedResult,
+	protocolDB storage.DB,
+	lockManager storage.LockManager,
+) *Core {
+	coreLogger := logger.With().
+		Str("component", "execution_data_core").
+		Str("execution_result_id", executionResult.ID().String()).
+		Str("block_id", executionResult.BlockID.String()).
+		Uint64("height", header.Height).
```

Comment:
```suggestion
		Uint64("height", header.Height).
		Uint64("view", header.View).
```

since we tend to have a high log volume, you could only log `block_id`, `height` and `view` once on construction in addition to `execution_result_id`. Subsequently, we only need to include the `execution_result_id` and the component name "execution_data_core" in order for logs to be uniquely identifiable. If you have the `execution_result_id`, which is included in every log, then you can obtain the remaining information from just searching for prior logs from "execution_data_core" with the same result ID.